Q1: A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
df = pd.read_csv('C:/Users/Lenovo/OneDrive/Desktop/assignments/Cutlets.csv')
df

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

Here we have to do is 2 tailed test

•h0 is µ1 = µ2

•h1 is µ1 ≠ µ2

In [2]:
stats.ttest_ind(df['Unit A'],df['Unit B'],alternative = 'two-sided')

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

As we can see here,

•pvalue is greater than 0.05.

•hence we failed to reject h0

•so we can conclude that there is no significant change in unit1 and unit 2.

Q2: A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch. Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

In [3]:
df1 =  pd.read_csv('C:/Users/Lenovo/OneDrive/Desktop/assignments/LabTAT.csv')
df1

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

•h0 is µ1 = µ2 = µ3 = µ4

•h1 is either any one of is not equal

•so here we have 2 do anova two sample test because it has more than 2 samples.

In [4]:
stats.f_oneway(df1['Laboratory 1'],df1['Laboratory 2'],df1['Laboratory 3'],df1['Laboratory 4'])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

• As we can see pvalue is 2.1156708949992414e-57 which is greater than the alpha value 0.05

• so failed to reject h0, hence can conclude that difference in average TAT among the different laboratories at 5% significance level.

Q3: Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

In [5]:
df2 =  pd.read_csv('C:/Users/Lenovo/OneDrive/Desktop/assignments/BuyerRatio.csv')
df2

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

Here as we can see,

•h0 is all proportions are equal

•h1 is all proportions are not equal

In [6]:
buyer_ratio = [[50,142,131,70],[435,1523,1356,750]]
print(buyer_ratio)

[[50, 142, 131, 70], [435, 1523, 1356, 750]]


In [7]:
stat, p_value, dof, expected = stats.chi2_contingency(buyer_ratio)
print("dof =%d " % dof)
print("pvalue :" ,p_value)

dof =3 
pvalue : 0.6603094907091882


Here as we can see,

• Pvalue is greater than alpha value, i.e 0.6603094907091882 > 0.05

• Hence we fail to reject h0, so male-female buyer rations are similar across regions.

Q4: TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences.

In [8]:
df3 =  pd.read_csv('C:/Users/Lenovo/OneDrive/Desktop/assignments/Costomer+OrderForm.csv')
df3

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [9]:
df3.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

H0 = no significant diffrence in defect % by centres

H1 = diffrence in the defect % by centres

In [10]:
df3.value_counts('Phillippines')

Phillippines
Error Free    271
Defective      29
dtype: int64

In [11]:
df3.value_counts('Indonesia')

Indonesia
Error Free    267
Defective      33
dtype: int64

In [12]:
df3.value_counts('Malta')

Malta
Error Free    269
Defective      31
dtype: int64

In [13]:
df3.value_counts('India')

India
Error Free    280
Defective      20
dtype: int64

In [16]:
#to test the contingency
cont = np.array([[271,267,269,280],[29,33,31,20]])
print(cont)
stat , p_value , dof , expected = stats.chi2_contingency(cont)
print('\np_value is',p_value)

[[271 267 269 280]
 [ 29  33  31  20]]

p_value is 0.2771020991233135


•Here also we can see that the pvalue is less than the alpha value

•So we reject H0.

•Hence there is no diffrence in the defective % variation by centres.